analyzer/aggregate_stats/aggregate_stats.go

```diff
@@ -0,0 +1,282 @@
+	"github.com/oasisprotocol/nexus/common"
+	"github.com/oasisprotocol/nexus/log"
+	"github.com/oasisprotocol/nexus/metrics"
+	"github.com/oasisprotocol/nexus/storage"
+)
+
+const (
+	aggregateStatsAnalyzerName = "aggregate_stats"
+
+	// Limit the number of stats aggregation windows to be computed in a single
+	// iteration.
+	// This limits database batch sizes in cases if the stats analyzer would be far behind
+	// latest indexed block (e.g. if it would not be enabled during initial sync).
+	// Likely never the case in normal operation.
+	statsBatchLimit = 1_000
+
+	// Name of the consensus layer.
+	layerConsensus = "consensus"
+
+	// Interval between stat computation iterations.
+	// The workers only computes new data and iteration with no updates are cheap
+	// so it's fine to run this frequently.
+	statsComputationInterval = 1 * time.Minute
+)
+
+// Layers that are tracked for aggregate stats.
+// XXX: Instead of hardcoding, these could also be obtained
+// by (periodically) querying the `runtime_related_transactions` table.
+var statsLayers = []string{
+	layerConsensus,
+	string(common.RuntimeEmerald),
+	string(common.RuntimeSapphire),
+	// RuntimeCipher.String(), // Enable once Cipher is supported by the indexer.
+}
+
+type aggregateStatsAnalyzer struct {
+	target storage.TargetStorage
+
+	logger  *log.Logger
+	metrics metrics.DatabaseMetrics
+}
+
+var _ analyzer.Analyzer = (*aggregateStatsAnalyzer)(nil)
+
+func (a *aggregateStatsAnalyzer) Name() string {
+	return aggregateStatsAnalyzerName
+}
+
+func NewAggregateStatsAnalyzer(target storage.TargetStorage, logger *log.Logger) (analyzer.Analyzer, error) {
+	logger.Info("starting aggregate_stats analyzer")
+	return &aggregateStatsAnalyzer{
+		target:  target,
+		logger:  logger.With("analyzer", aggregateStatsAnalyzerName),
+		metrics: metrics.NewDefaultDatabaseMetrics(aggregateStatsAnalyzerName),
+	}, nil
+}
+
+func (a *aggregateStatsAnalyzer) Start(ctx context.Context) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		a.aggregateStatsWorker(ctx)
+	}()
+
+	wg.Wait()
+}
+
+func (a *aggregateStatsAnalyzer) writeToDB(ctx context.Context, batch *storage.QueryBatch, opName string) error {
+	timer := a.metrics.DatabaseTimer(a.target.Name(), opName)
+	defer timer.ObserveDuration()
+
+	if err := a.target.SendBatch(ctx, batch); err != nil {
+		a.metrics.DatabaseCounter(a.target.Name(), opName, "failure").Inc()
+		return err
+	}
+	a.metrics.DatabaseCounter(a.target.Name(), opName, "success").Inc()
+
+	return nil
+}
+
+func (a *aggregateStatsAnalyzer) aggregateStatsWorker(ctx context.Context) {
+	statsComputations := []*statsComputation{}
+
+	// Compute 5-minute tx volume stats every 5 minutes for all layers.
+	for _, layer := range statsLayers {
+		statsComputations = append(statsComputations, &statsComputation{
+			name:                "min5_tx_volume_" + layer,
+			layer:               layer,
+			dbTable:             "stats.min5_tx_volume",
```

Comment:
Can we call this `outputTable` and `outputColumn`? To differentiate from the input side of things (which also uses the DB).

Reply:
Good suggestion, thanks! Updated.